app/services/puppet_fact_parser.rb

```diff
@@ -55,6 +60,7 @@ def environment
   end
 
   def architecture
+    return @architecture if @architecture.present?
```

Comment:
If I'm not wrong you could use `attr_accessor` here to avoid these guard clauses.